src/main/java/com/myaudiolibrary/web/controller/ArtistController.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    @GetMapping(
+            value = "",
+            params = {"page", "size", "sortProperty", "sortDirection"},
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public String artistPagedList(
+            final ModelMap model,
+            @RequestParam(value = "page", defaultValue = "0") Integer page,
+            @RequestParam(value = "size", defaultValue = "10") Integer size,
+            @RequestParam(value = "sortProperty", defaultValue = "name") String sortProperty,
+            @RequestParam(value = "sortDirection", defaultValue = "ASC") Sort.Direction sortDirection) {
+        checkParams(page, size, sortDirection);
+        Pageable pageable = PageRequest.of(page, size, Sort.by(sortDirection, sortProperty));
+        Page<Artist> artistsPage = artistRepository.findAll(pageable);
+        model.put("artists", artistsPage);
+        model.put("pageNumber", page + 1);
+        // IL FAUT DISABLE LE BOUTTON SI ON EST SUR LA PREMIERE PAGE !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!s!!!!!!!!!!!!!!!!!
+        model.put("previousPage", page - 1);
+        model.put("nextPage", page + 1);
+        return "listeArtists";
+    }
+
+    @GetMapping(value = "/new")
+    public String newArtist(final ModelMap model) {
+        model.put("artist", new Artist());
+        return "detailArtist";
+    }
+
+    @PostMapping(value = "", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public RedirectView createArtist(Artist artist) throws InstanceAlreadyExistsException {
+        if(artistRepository.existsByName(artist.getName())) {
+            throw new InstanceAlreadyExistsException("L'artiste " + artist.getName() + " existe déjà.");
+        }
+        artist = artistRepository.save(artist);
+        return new RedirectView("/artists/" + artist.getId());
+    }
+
+    @PutMapping(value = "/{id}", consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+    public RedirectView saveArtist(Artist artist, @PathVariable Long id) throws EntityNotFoundException {
+        if (!artistRepository.existsById(id)) {
+            throw new EntityNotFoundException("L'artiste d'identifiant " + id + " n'existe pas.");
+        }
+        if (!id.equals(artist.getId())) {
+            throw new IllegalArgumentException("Requete invalide.");
+        }
+        artistRepository.save(artist);
+        return new RedirectView("/artists/" + id);
+    }
+
+    @DeleteMapping(value = "/{id}")
+    @Transactional
+    public RedirectView deleteArtist(@PathVariable Long id) throws EntityNotFoundException {
+        Optional<Artist> artist = artistRepository.findById(id);
+        if(artist.isEmpty()) {
+            throw new EntityNotFoundException("L'artiste d'identifiant " + id + " n'existe pas.");
+        }
+        albumRepository.deleteByArtist(artist);
+        artistRepository.deleteById(id);
+        return new RedirectView("/artists?page=0&size=10&sortProperty=name&sortDirection=ASC");
```

Comment:
Pas la peine d'ajouter tous les paramètres, ils ont une valeur par défaut
```suggestion
        return new RedirectView("/artists");
```